base/sysinfo.jl

```diff
@@ -567,4 +568,25 @@ function which(program_name::String)
 end
 which(program_name::AbstractString) = which(String(program_name))
 
+"""
+    username() -> String
+
+Return the username for the current user. If the username cannot be determined
+or is empty, this function throws an error.
+
+To retrieve a username that is overridable via an environment variable,
+e.g., `USER`, consider using
+```julia
+user = get(ENV, "USER", username())
+```
+
+!!! compat "Julia 1.11"
+    This function requires at least Julia 1.11.
+"""
+function username()
+    pw = Libc.getpw()
+    isempty(pw.username) || return pw.username
+    throw(ErrorException("username was empty"))
```

Comment:
I think it's a bit clearer to put the `return` statement last:
```suggestion
    isempty(pw.username) && throw(ErrorException("empty username"))
    return pw.username
```